plutus-core/plutus-ir/src/PlutusIR/Transform/Rename.hs

```diff
@@ -58,9 +59,8 @@ Two problems arise:
 -}
 
 instance PLC.HasUniques (Term tyname name uni fun ann) => PLC.Rename (Term tyname name uni fun ann) where
-    -- TODO: the Plutus Core codebase uses marking in order to prevent clashing with existing
-    -- free variables. Should we do the same here?
-    rename = PLC.runRenameT . renameTermM
+    -- See Note [Marking]
+    rename = through markNonFreshTerm >=> PLC.runRenameT . renameTermM
```

Comment:
Hmm, I suggested that Jakub try this in the past and it made things fail. I guess it doesn't here.

Reply:
Yes, see my comment above.